### PR TITLE
GATK4 first round without MuTect1 and indel realignment

### PR DESCRIPTION
**Pull request by szilvajuhos**
Also have a look at the new container structure. I am trying to accommodate nf-core guidelines. alleleCount and ASCAT needs new bioconda recipes, but most of the other tools are in a collated (relatively big) container including GATK4, igvtools, etc.

### CODE REVIEW

File: annotate.nf [hunk 1]
```diff
@@ -73,23 +73,23 @@ vcfToAnnotate = Channel.create()
 vcfNotToAnnotate = Channel.create()
 
 if (annotateVCF == []) {
+// by default we annotate both germline and somatic results that we can find in the VariantCalling directory
```
Comment (MaxUlysse) [inline annotate.nf]: In fact, we annote all available vcfs by default, so it's not really a question of germline/somatic, but really more a question of which tools was run

File: containers/sarek/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,9 @@
-FROM debian:8.9
+FROM nfcore/base:latest
 
 LABEL \
-  author="Maxime Garcia" \
+  authors="Maxime.Gracia@scilifelab.se, Szilveszter.Juhos@scilifelab.se" \
```
Comment (MaxUlysse) [inline containers/sarek/Dockerfile]: typo in my name.

File: containers/sarek/Dockerfile [hunk 3]
```diff
@@ -1,18 +1,9 @@
...
   description="Image with tools used in Sarek" \
-  maintainer="maxime.garcia@scilifelab.se"
+  maintainers="Maxime.Gracia@scilifelab.se, Szilveszter.Juhos@scilifelab.se"
```
Comment (MaxUlysse) [inline containers/sarek/Dockerfile]: If I remember well, we can use whichever label we want, but `maintainer` is meant to stay that way, because it's a port of the deprecated instruction MAINTAINER.
I do think we can use:
```
maintainer="Maxime Garcia <maxime.garcia@scilifelab.se>, Szilveszter Juhos <Szilveszter.Juhos@scilifelab.se>"
```

File: containers/vcfanno/to_build [hunk 4]
```diff
@@ -0,0 +1,5 @@
+docker login szilvajuhos
+docker build -t szilvajuhos/sarek-vcfanno:latest .
+docker images
+docker push szilvajuhos/sarek-vcfanno:latest
+singularity pull docker://szilvajuhos/sarek-vcfanno:latest
```
Comment (MaxUlysse) [inline containers/vcfanno/to_build]: Do we really need this script in this repo ?

File: germlineVC.nf [hunk 5]
```diff
@@ -310,20 +307,15 @@ process RunHaplotypecaller {
...
-  --disable_auto_index_creation_and_locking_when_reading_rods \
-  -o ${intervalBed.baseName}_${idSample}.g.vcf
+  gatk-launch --java-options "-Xmx${task.memory.toGiga()}g -Xms6000m -XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10" \
+      HaplotypeCaller \
+      -R ${genomeFile} \
+      -I ${bam} \
+      -L ${intervalBed} \
+  		--dbsnp ${dbsnp} \
+  		-O ${intervalBed.baseName}_${idSample}.g.vcf 	\
+			--emit-ref-confidence GVCF
```
Comment (MaxUlysse) [inline germlineVC.nf]: strange indentation

File: lib/QC.groovy [hunk 6]
```diff
@@ -59,14 +59,14 @@ class QC {
 // Get GATK version
   static def getVersionGATK() {
     """
-    echo "GATK version"\$(java -jar \$GATK_HOME/GenomeAnalysisTK.jar --version 2>&1) > v_gatk.txt
+		gatk-launch ApplyBQSR --help 2>&1| awk -F/ '/java/{for(i=1;i<=NF;i++){if(\$i~/gatk4/){sub("gatk4-","",\$i);print \$i>"v_gatk.txt"}}}'
```
Comment (MaxUlysse) [inline lib/QC.groovy]: I think we can work out the regex in the Python script instead of doing it here, it'll make more sense

**Comment by MaxUlysse** [comment]
Quite an impressive work.
Well done @szilvajuhos \o/

File: lib/QC.groovy (hunk 6 above)
Comment (szilvajuhos) [reply lib/QC.groovy]: the gatk-launch is GATK-provided, I do not want to fiddle with that. OTOH it would be nice if they would have a --version option :/

File: containers/vcfanno/to_build (hunk 4 above)
Comment (szilvajuhos) [reply containers/vcfanno/to_build]: No, of course not

File: lib/QC.groovy (hunk 6 above)
Comment (MaxUlysse) [reply lib/QC.groovy]: I'll look more if there's something similar with the new GATK

File: containers/sarek/environment.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  - conda-forge
+  - defaults
+
+dependencies:
+  - bcftools=1.8
+  - bwa=0.7.17
+  - conda-forge::openjdk=8.0.144 # Needed for FastQC docker - see bioconda/bioconda-recipes#5026
+  - fastqc=0.11.7
+  - freebayes=1.2.0
+  - gatk4=4.0.3.0
```
Comment (MaxUlysse) [inline containers/sarek/environment.yml]: You can use the 4.0.4.0, the executable is back to being `gatk` and not `gatck-launch` anymore

File: containers/sarek/environment.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  - htslib=1.7
```
Comment (MaxUlysse) [inline containers/sarek/environment.yml]: You should use the 1.8 here.
If I remember well, `htslib`, `bcftools` and `samtools` can all have the same version

File: containers/sarek/environment.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  - igvtools=2.3.93
+  - manta=1.3.0
```
Comment (MaxUlysse) [inline containers/sarek/environment.yml]: Since we're updating, we can try the 1.4.0

Comment (szilvajuhos) [reply containers/sarek/environment.yml]: OK, done

File: containers/sarek/environment.yml (hunk 8 above)
Comment (szilvajuhos) [reply containers/sarek/environment.yml]: OK, done but will check since I got a feeling that 1.8 has compatibility issues

File: containers/sarek/environment.yml (hunk 7 above)
Comment (szilvajuhos) [reply containers/sarek/environment.yml]: Fine, will change the name in processes as well. In fact we have 4.0.6.0 also

File: lib/QC.groovy (hunk 6 above)
Comment (szilvajuhos) [reply lib/QC.groovy]: My feeling is that it is still the easiest way to have the version :/

File: containers/sarek/environment.yml (hunk 7 above)
Comment (MaxUlysse) [reply containers/sarek/environment.yml]: Even better ;-)

File: lib/QC.groovy (hunk 6 above)
Comment (MaxUlysse) [reply lib/QC.groovy]: I was thinking more of removing the awk part, and do the regex in the [python script](https://github.com/SciLifeLab/Sarek/blob/master/bin/scrape_tool_versions.py)

File: containers/sarek/environment.yml (hunk 8 above)
Comment (MaxUlysse) [reply containers/sarek/environment.yml]: OK, strange, but good to know if you can confirm that

Comment (apeltzer) [reply containers/sarek/environment.yml]: There is even 1.9 out already! You could already skip 1.8 ... 
https://github.com/samtools/samtools/releases/

File: lib/QC.groovy (hunk 6 above)
Comment (szilvajuhos) [reply lib/QC.groovy]: I see, sure, we can refactor it for the rest of the software later also.

File: configuration/uppmax-localhost.config [hunk 10]
```diff
@@ -14,7 +14,7 @@ env {
 params {
   genome_base = params.genome == 'GRCh37' ? '/sw/data/uppnex/ToolBox/ReferenceAssemblies/hg38make/bundle/2.8/b37' : params.genome == 'GRCh38' ? '/sw/data/uppnex/ToolBox/hg38bundle' : 'References/smallGRCh37'
   singleCPUMem  = 8.GB
-  totalMemory = 104.GB // change to 240 on irma
+  totalMemory = 92.GB // change to 240 on irma
```
Comment (MaxUlysse) [inline configuration/uppmax-localhost.config]: Why did you change to 92?

File: containers/sarek/Dockerfile [hunk 11]
```diff
@@ -1,18 +1,10 @@
...
-  STRELKA_INSTALL_PATH=/opt/strelka
-
-# Build container
-ADD \
-  build.sh /usr/bin/
-RUN \
-  build.sh
+COPY environment.yml /
+RUN conda env update -n root -f /environment.yml && conda clean -a
+ENV PATH /opt/conda/bin:$PATH
```
Comment (MaxUlysse) [inline containers/sarek/Dockerfile]: we can get rid of this last line:
https://gitter.im/nf-core/Lobby?at=5b59f41bd2f0934551d30d5d

File: containers/sarek/environment.yml [hunk 12]
```diff
@@ -0,0 +1,24 @@
+# You can use this file to create a conda environment for this pipeline:
+# conda env create -f environment.yml
+name: sarek-core
```
Comment (MaxUlysse) [inline containers/sarek/environment.yml]: We should specify a version here
so I would go for `sarek-core-dev` or `sarek-core-2.1`

File: configuration/uppmax-localhost.config (hunk 10 above)
Comment (szilvajuhos) [reply configuration/uppmax-localhost.config]: by mistake

File: containers/sarek/Dockerfile (hunk 11 above)
Comment (szilvajuhos) [reply containers/sarek/Dockerfile]: tried, and was not working as expected, so I prefer to leave it as it is now, and improve when needed

File: containers/sarek/environment.yml (hunk 12 above)
Comment (szilvajuhos) [reply containers/sarek/environment.yml]: can we just leave as `sarek` ?

File: containers/sarek/Dockerfile (hunk 11 above)
Comment (szilvajuhos) [reply containers/sarek/Dockerfile]: we will keep now with the ENV
